### PR TITLE
Modbus nan values

### PR DESCRIPTION
**Pull request by joanwa**
## Proposed change

add documentation to optional modbus parameter: nan_value

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/90800
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/core/issues/69656
- This PR fixes or closes issue: fixes #

## Checklist

- [ x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -1001,6 +1001,10 @@ sensors:
       description: The maximum allowed value of a sensor. If value > max_value --> max_value. Can be float or integer
       required: false
       type: float
+    nan_value:
+      description: The NaN value if the sensor has an explicit NaN return. Can be float, integer or in hex notation
+      required: false
+      type: int
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: It does not tell anything about the purpose of the parameter and what it does.

**Review by janiversen — CHANGES_REQUESTED** [review]

File: source/_integrations/modbus.markdown (hunk 1 above)
Comment (joanwa) [reply source/_integrations/modbus.markdown]: Better like this?
If a modbus sensor has an explicit NaN value, this optional parameter gives the ability to specify it, such that the modbus sensor returns "unavailable" when NaN is returned

**Comment by janiversen** [comment]
However the PR is marked as draft, meaning we cannot review and approve it.

**Comment by janiversen** [comment]
Please correct the errors, and mark as ready for review. Remark without this the corresponding PR in core will not be approved.

File: source/_integrations/modbus.markdown [hunk 2]
```diff
@@ -1013,6 +1013,10 @@ sensors:
       description: The maximum allowed value of a sensor. If value > max_value --> max_value. Can be float or integer
       required: false
       type: float
+    nan_value:
+      description: If a modbus sensor has a vendor defined NaN value, this value can be set here as hex string (e.g. `0xFFFFFFFF`) and, if triggered, the sensor becomes `unavailable`.
```
Comment (janiversen) [inline source/_integrations/modbus.markdown]: In the code PR it says that the Na value can be a int, float or he. value. This should be matched here.

**Review by janiversen — CHANGES_REQUESTED** [review]

File: source/_integrations/modbus.markdown (hunk 2 above)
Comment (joanwa) [reply source/_integrations/modbus.markdown]: the PR is string input also, first version with int/float/hex had issues. Should be consistent with documentation here

**Comment by janiversen** [comment]
Let's see once the core PR is ready.

File: source/_integrations/modbus.markdown (hunk 2 above)
Comment (janiversen) [reply source/_integrations/modbus.markdown]: The e.g. does not correspond to the code, you demand a specific format, so please inform the user clearly.

Comment (joanwa) [reply source/_integrations/modbus.markdown]: you're right, thanks!

**Comment by janiversen** [comment]
??? you are changing 40 files, seems you did not update your repo,

**Comment by joanwa** [comment]
> ??? you are changing 40 files, seems you did not update your repo,

sorry, synched by accident with current, not next

**Review by janiversen — DISMISSED** [review]
LGTM, thanks

File: source/_integrations/modbus.markdown [hunk 3]
```diff
@@ -1013,6 +1013,10 @@ sensors:
       description: The maximum allowed value of a sensor. If value > max_value --> max_value. Can be float or integer
       required: false
       type: float
+    nan_value:
+      description: If a modbus sensor has a defined NaN value, this value can be set as hex string starting with `0x` containing one or more bytes (e.g. `0xFFFF` or `0x80000000`) or provided as integer directly. If triggered, the sensor becomes `unavailable`. Please note that the hex to int conversion for `nan_value` does currently not obey home-assistants modbus encoding using the `data_type`, `structure` or `swap` arguments.
```
Comment (frenck) [inline source/_integrations/modbus.markdown]:
```suggestion
      description: If a Modbus sensor has a defined NaN value, this value can be set as a hex string starting with `0x` containing one or more bytes (for example, `0xFFFF` or `0x80000000`) or provided as an integer directly. If triggered, the sensor becomes `unavailable`. Please note that the hex to int conversion for `nan_value` does currently not obey home-assistants Modbus encoding using the `data_type`, `structure`, or `swap` arguments.
```

**Review by frenck — APPROVED** [review]
Thanks, @joanwa 👍 

../Frenck

**Pull request merged**